source/observables.txt

```diff
@@ -0,0 +1,243 @@
+- Customizes the ``onNext()`` method to print each retrieved document,
+  which are ``Document`` instances
+- Customizes the ``onError()`` method to print any errors
+- Customizes the ``onComplete()`` methods to print a message after all
+  the results from the ``Observable`` are processed
+
+.. io-code-block::
+   :copyable:
+
+   .. input:: /includes/observables.scala
+      :language: scala
+      :start-after: start-read-observable
+      :end-before: end-read-observable
+      :dedent:
+
+   .. output::
+      :language: console
+      :visible: false
+
+      Iterable((_id, ...), ..., (name,BsonString{value='Koliba Restaurant'}),
+      (restaurant_id,BsonString{value='40812870'}))
+      Iterable((_id, ...), ..., (name,BsonString{value='Bohemian Beer Garden'}),
+      (restaurant_id,BsonString{value='41485121'}))
+      Iterable((_id,...), ..., (name,BsonString{value='Hospoda'}),
+      (restaurant_id,BsonString{value='41569184'}))
+      Iterable((_id,...), ..., (name,BsonString{value='Olde Prague Tavern'}),
+      (restaurant_id,BsonString{value='41711983'}))
+      Processed all results
+
+Access Write Operation Results
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To access data retrieved by a write operation, create an ``Observable[T]``
+to store the operation results. Then, subscribe to the observable and
+override the ``Observer`` class callback methods to process the results.
+
+This example inserts documents into the ``restaurants`` database
+in which the ``cuisine`` value is ``"Nepalese"``. To retrieve
+and process results, the example assigns an ``Observable[InsertManyResult]``
+to the operation and performs the following actions:
+
+- Calls the ``subscribe()`` method to subscribe to the ``Observable``
+  and passes an ``Observer`` as a parameter
+- Customizes the ``onNext()`` method to print the result of the insert
+  operation, returned as an ``InsertManyResult``
+- Customizes the ``onError()`` method to print any errors
+- Customizes the ``onComplete()`` methods to print a message after all
+  the results from the ``Observable`` are processed
+
+.. io-code-block::
+   :copyable:
+
+   .. input:: /includes/observables.scala
+      :language: scala
+      :start-after: start-read-observable
+      :end-before: end-read-observable
+      :dedent:
+
+   .. output::
+      :language: console
+      :visible: false
+
+      AcknowledgedInsertManyResult{insertedIds={0=BsonObjectId{value=...},
+      1=BsonObjectId{value=...}}}
+      Processed all results
+
+Use Lambda Functions to Process Results
+---------------------------------------
+
+Instead of explicitly overriding the callback functions from the
+``Observer`` class, you can use lambda functions to concisely
+process operation results. These functions allow you to use the
+``=>`` arrow notation to customize the implementation of ``onNext()``,
+``onError()``, and ``onComplete()``.
+
+.. tip::
+
+   To learn more about lambda, or anonymous, functions,
```

Comment:
S: This reads bit more smoothly for me but also if you have other ideas feel free to edit:

```suggestion
   To learn more about lambda functions, also known as anonymous functions,
```